src/main/java/model/IProduct.java

```diff
@@ -0,0 +1,9 @@
+package model;
+
+public interface IProduct {
```

Comment:
Hier streiten sich die Geister - in der Uni hatte ich auch noch gelernt, dass man ein "I" vor die Klasse packt, um ein Interface zu signalisieren.

Heutzutage ist es unüblich, weil es wenig Mehrwert bringt.
Schau dir mal diese Antwort an: https://stackoverflow.com/a/2814831/11515036

tl;dr 
Nenn die Interfaces einfach wie sie sind. Z.B. "Product" und spezifiziere die Unterklassen.
Z.B. "ElectronicProduct", "HealthProduct" etc